### PR TITLE
fix(treesitter_context): underline content if `transparent_background` is true

### PR DESCRIPTION
**Pull request by joshmedeski**
Updates the Treesitter Context plugin to use a transparent background if transparent_background was set.

### CODE REVIEW

**Comment by nullchilly** [comment]
treesitter_context integration is not enabled by default and thus support transparent background out of the box

The reason the integration still exists is https://github.com/catppuccin/nvim/issues/445

**Comment by joshmedeski** [comment]
Okay, I won't enable the integration so I can get transparency.

Do you think I should update this PR to add a code comment mentioning that enabling this integration makes it distinguishable when transparency is enabled? That is unclear when looking at the source-code. It just appears to be a missing feature.

If not, feel free to just close the PR.

**Comment by nullchilly** [comment]
I think we can change it to use `TreesitterContextBottom` if transparency is enabled

**Comment by nullchilly** [comment]
I think we can define `hi TreesitterContextBottom gui=underline guisp=Grey` when transparency is enabled

image

**Review by nullchilly — CHANGES_REQUESTED** [review]

File: lua/catppuccin/groups/integrations/treesitter_context.lua [hunk 1]
```diff
@@ -1,8 +1,11 @@
 local M = {}
 
 function M.get()
+	local transparent_background = require("catppuccin").options.transparent_background
+	local bg_highlight = transparent_background and "NONE" or C.mantle
 	return {
-		TreesitterContext = { bg = C.mantle, fg = C.text },
+		TreesitterContext = { bg = bg_highlight, fg = C.text },
+		TreesitterContextBottom = { gui = "underline", guisp = C.dim },
```
[the remark this answers is not in the document]
Comment (joshmedeski) [reply lua/catppuccin/groups/integrations/treesitter_context.lua]: Sounds good, I updated it.

**Comment by nullchilly** [comment]
LGTM!

**Review by mrtnvgr — APPROVED** [review]
LGTM!

**Pull request merged**